java/ql/lib/semmle/code/java/dataflow/DataFlowFiltering.qll

```diff
@@ -0,0 +1,125 @@
+/**
+ * Provides Java-specific dataflow config wrappers that apply alert filtering.
+ */
+
+private import semmle.code.java.AlertFiltering
+private import semmle.code.java.dataflow.DataFlow
+
+// TODO: The FilteredConfig and FilteredStateConfig modules should be additionally parameterized
+// with Node and Location types so that they can be shared across languages instead of being
+// specific to Java only.
```

Comment:
Why not just do this right away? I.e., move this module into the `dataflow` pack. I suggest putting it after the `MergePathGraph3` module at https://github.com/github/codeql/blob/main/shared/dataflow/codeql/dataflow/DataFlow.qll#L760.

Reply:
When I tried to do it the first time I couldn't for some reason get it to work.  But you are right, I think it is time for me to put more effort in making these modules language-agnostic.

Reply:
I updated the PR, and now `FilteredConfig` and `FilteredStateConfig` are both shared across languages.  Thanks for the suggestion to add those modules within `DataFlowMake`—that made it really easy to ensure that `Node` and `Location` are parameterized correctly.